main.nf

```diff
@@ -354,6 +358,30 @@ if(!params.bed){
     }
 }
 
+/*
+ * PREPROCESSING - Prepare genome size table for ATAC-seq
+ */
+ if(params.ATACseq){
+     process makeGenomeSizeTable {
+
+         publishDir path: { params.saveReference ? "${params.outdir}/reference_genome" : params.outdir },
+                    saveAs: { params.saveReference ? it : null }, mode: 'copy'
+
+         input:
+         file index from bwa_index_genomesizetable.collect()
+
+         output:
+         file "*.txt" into genomeSizeTable
+
+         script:
+         """
+         ln -s ${index}/genome.fa genome.fa
```

Comment:
Why do you need to softlink this? Can we not do this in a single command?

```bash
samtools faidx ${index}/genome.fa | cut -f1,2 > genomeSizeTable.txt
```

Reply:
Because when I ran `samtools faidx ${index}/genome.fa` it gave an error that `fai cannot be build under BWAIndex`

Reply:
If you specify this genome.fa as an input shouldn't Nextflow make this softlink for you ?

Reply:
I just spent longer than I should have looking into this. You are completely correct: _there is no way to specify the output file for `samtools faidx`_. That is ridiculous.

So yeah, you can't create the file inside the channel directly so your current approach is indeed the best.

Reply:
There is a softlinked dir for `BWAIndex`. But I cannot write in that dir therefore `samtools faidx ${index}/genome.fa` does not work.

Reply:
Aaaah, sorry, I haven't look where this `genome.fa` came from.

We're doing that in Sarek:
https://github.com/SciLifeLab/Sarek/blob/0fd73f78ed80c86843823835098074d1a62b851d/buildReferences.nf#L118-L134

Reply:
Yeah - here the channel is the directory though, not the index files. Samtools insists on writing the index file in the same directory as the fasta file. Because of how nextflow staging works, there isn't write permissions in the directory so it can't make the file there.

Reply:
https://github.com/SciLifeLab/Sarek/blob/0fd73f78ed80c86843823835098074d1a62b851d/buildReferences.nf#L164-L179